build/helpers/file-includes-all-subdirectories-as-named-exports.ts

```diff
@@ -0,0 +1,43 @@
+import * as path from "path";
+import * as fs from "fs";
+
+/**
+ * Verifies all components in source directory are exported
+ */
+export function includesAllSubdirectoriesAsNamedExports(indexFile: string): any {
+    /**
+     * Get the folders in the indexFile directory to compare to export listing
+     */
+    const directoryPath: string = path.dirname(indexFile);
+    const components: string[] = [];
+    const sourceDirectories: string[] = fs.readdirSync(directoryPath).map(
+        (name: string) => path.join(directoryPath, name)).filter(
+            (source: string) => fs.statSync(source).isDirectory());
```

Comment:
I think you should just `require` the path and then test all the keys of that object - that seems less fragile than regexing the contents of the file.

Reply:
Can't do this due to the fact with export differently in different repos. I left notes about this in the file.

Reply:
Can you describe the issue you're having? Why does differing export syntax prevent us from taking that approach?

Reply:
So, here is what the export data looks like across packages:
```
/**
 * Example formatting from fast-components-react-base
 */
import Foo from "./foo";
export { Foo };
export * from "./foo";

import Baz from "./baz";
export { Baz };
export * from "./baz";


/**
 * Example formatting from fast-component-react-msft
 */
export * from "./bar";
```
so, requiring like this:
```
const foo: any = require(indexFile);
console.log(Object.keys(foo));
```
results in `[ 'Foo', 'default', 'Baz' ]`

See the issue?

Reply:
Correct - sorry I thought supplying the full detail would be more helpful.

Reply:
In a quick prototype it *did* have those keys available. Given the following export file:

```
const Foo = "Foo";
const bar = "bar";

export default Foo;
export { bar };
```

The keys on the require object are "['default', 'bar']".

If you're getting something different than lets work together to get it figured out - I think testing the actual export object will be more reliable long-term.

Reply:
@nicholasrice  sure ping me whenever. To be clear, the exports in `fast-react-component-react` looks like this _only_
```
export * from "./button";
export * from "./call-to-action";
export * from "./caption";
```

Reply:
Should be all good now - thanks for the help